src/Telemetry/Library/TelemetryManagerInstance.cs

```diff
@@ -417,8 +420,14 @@ private void CheckAndRunSafe(Action action, bool initializingOrTearingDown = fal
                 .Where(address => address != null && !string.IsNullOrWhiteSpace(address) && !address.StartsWith("000000"))
                 .FirstOrDefault();
 
-        private bool GetTelemetryOptOut(TelemetryManagerConfig configuration) =>
-            Environment.GetEnvironmentVariable(configuration.TelemetryOptOutVariableName) == "1";
+        [SuppressMessage("StyleCop.CSharp.SpacingRules", "SA1009:ClosingParenthesisMustBeSpacedCorrectly", Justification = "Easier to read the logic.")]
+        private bool GetTelemetryOptIn(TelemetryManagerConfig configuration) =>
+            (
+                (configuration.DefaultTelemetryConsent == ConsentKind.OptedIn)
+                ||
+                (Environment.GetEnvironmentVariable(configuration.TelemetryOptInVariableName) == "1")
+            )
+            && !(Environment.GetEnvironmentVariable(configuration.TelemetryOptOutVariableName) == "1");

```

Comment:
I'm not sure I understand the motivation behind the opt-out env variable having precedence. Shouldn't it be something like:
```
telemetryConsent =
    consentKind == OptIn && Environment.GetEnvironmentVariable(configuration.TelemetryOptInVariableName) == "1"
    ||
    !(Environment.GetEnvironmentVariable(configuration.TelemetryOptOutVariableName) == "1")
```

Reply:
The opt-out env var takes precedence just in an odd situation in which both environment options are set.
Normally the user would only want (and need) to specify one of them to deviate from the default consent.